### PR TITLE
feat: [UPC-6298] Audio level indicator when using media stream for mobile 2-way audio

### PR DESCRIPTION
**Pull request by UI-RayanWang**
[Ticket](https://ubiquiti.atlassian.net/browse/UPC-6298)

### CODE REVIEW

File: sdk/objc/native/src/unifi_capture_post_processing.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  virtual ~UnifiCapturePostProcessing() override {}
+
+  void Initialize(int sample_rate_hz, int num_channels) override;
+  void Process(AudioBuffer* audio) override;
+  std::string ToString() const override;
+  void SetRuntimeSetting(AudioProcessing::RuntimeSetting setting) override;
+  void UpdateAudioIndicator(int average, int peak) override;
+
+ private:
+  id<RTC_OBJC_TYPE(RTCAudioProcessing)> audio_processing_;
```
Comment (AtinYehUI) [inline sdk/objc/native/src/unifi_capture_post_processing.h:26]: iOS memory management (ARC), variable default strong reference. 

virtual ~UnifiCapturePostProcessing() override { audio_processing_ = nil; }
or 
__weak  id<RTC_OBJC_TYPE(RTCAudioProcessing)> audio_processing_;

Comment (UI-RayanWang) [reply sdk/objc/native/src/unifi_capture_post_processing.h:26]: ARC should auto release it when this object is destroyed, shouldn't it?

Comment (UI-RayanWang) [reply sdk/objc/native/src/unifi_capture_post_processing.h:26]: > iOS memory management (ARC), variable default strong reference.
> 
> virtual ~UnifiCapturePostProcessing() override { audio_processing_ = nil; } or __weak id<RTC_OBJC_TYPE(RTCAudioProcessing)> audio_processing_;

ARC should auto release it when this object is destroyed, shouldn't it?

Comment (AtinYehUI) [reply sdk/objc/native/src/unifi_capture_post_processing.h:26]: reference google webrtc objc class RTCPeerConnection+Private.h suggest use __weak 
Screenshot 2023-11-01 at 12 59 55 PM

**Review by AtinYehUI — APPROVED** [review]

File: sdk/objc/native/src/unifi_capture_post_processing.h (hunk 1 above)
Comment (UI-RayanWang) [reply sdk/objc/native/src/unifi_capture_post_processing.h:26]: > reference google webrtc objc class RTCPeerConnection+Private.h suggest use __weak Screenshot 2023-11-01 at 12 59 55 PM

What's the difference between `id<RTC_OBJC_TYPE(RTCAudioProcessing)>` and `RTCAudioProcessing*`?
As I see in for example the `objc_video_decoder_factory.h/.mm`, looks like the "set to nil" operation is unnecessary?
https://github.com/ubiquiti/ubnt_libjingle/blob/feat/iOS-audio-indicator/sdk/objc/native/src/objc_video_decoder_factory.h#L35

**Review by ogsts — APPROVED** [review]

**Pull request merged**